src/bun.js/ipc.zig

```diff
@@ -156,17 +156,162 @@ pub const IPCData = struct {
+        }
+        const VersionPacket = extern struct {
+            type: IPCMessageType align(1) = .Version,
+            version: u32 align(1) = ipcVersion,
+        };
+
+        if (Environment.allow_assert) {
+            this.has_written_version = 1;
+        }
+        const bytes = comptime std.mem.asBytes(&VersionPacket{});
+        // enqueue to be sent after connecting
+        var list = this.outgoing.list.listManaged(bun.default_allocator);
+        list.appendSlice(bytes) catch bun.outOfMemory();
+        if (this.connected) {
+            this.processSend();
+        }
+    }
+
+    pub fn serializeAndSend(this: *NamedPipeIPCData, globalThis: *JSGlobalObject, value: JSValue) bool {
+        if (Environment.allow_assert) {
+            std.debug.assert(this.has_written_version == 1);
+        }
+
+        const serialized = value.serialize(globalThis) orelse return false;
+        defer serialized.deinit();
+
+        const size: u32 = @intCast(serialized.data.len);
+        log("serializeAndSend {d}", .{size});
+
+        const payload_length: usize = @sizeOf(IPCMessageType) + @sizeOf(u32) + size;
+
+        this.outgoing.list.ensureUnusedCapacity(bun.default_allocator, payload_length) catch @panic("OOM");
+        const start_offset = this.outgoing.list.len;
+
+        this.outgoing.list.writeTypeAsBytesAssumeCapacity(u8, @intFromEnum(IPCMessageType.SerializedMessage));
+        this.outgoing.list.writeTypeAsBytesAssumeCapacity(u32, size);
+        this.outgoing.list.appendSliceAssumeCapacity(serialized.data);
+
+        std.debug.assert(this.outgoing.list.len == start_offset + payload_length);
+
+        if (start_offset == 0) {
+            std.debug.assert(this.outgoing.cursor == 0);
+            if (this.connected) {
+                this.processSend();
+            }
+        }
+
+        return true;
+    }
+
+    pub fn close(this: *NamedPipeIPCData, comptime Context: type) void {
+        if (this.server.loop != null) {
+            _ = uv.uv_close(@ptrCast(&this.pipe), NewNamedPipeIPCHandler(Context).onServerClose);
+        } else {
+            _ = uv.uv_close(@ptrCast(&this.pipe), NewNamedPipeIPCHandler(Context).onClose);
+        }
+    }
+
+    pub fn configureServer(this: *NamedPipeIPCData, comptime Context: type, instance: *Context, named_pipe: []const u8) c_int {
+        log("configureServer", .{});
+        const ipc_pipe = &this.server;
+
+        var errno = uv.uv_pipe_init(uv.Loop.get(), ipc_pipe, 0);
+        if (errno != 0) {
+            return errno;
+        }
+        ipc_pipe.data = @ptrCast(instance);
+        errno = uv.uv_pipe_bind2(ipc_pipe, named_pipe.ptr, named_pipe.len, 0);
+        if (errno != 0) {
+            return errno;
+        }
+        errno = uv.uv_listen(@ptrCast(ipc_pipe), 0, NewNamedPipeIPCHandler(Context).onNewClientConnect);
+        if (errno != 0) {
+            return errno;
+        }
+
+        uv.uv_pipe_pending_instances(ipc_pipe, 1);
```

Comment:
This should be a function on ipc_pipe, with a clearer name. Like `ipc_pipe.setPendingInstancesCount(1)`